importer/api/__init__.py

```diff
@@ -36,17 +37,22 @@ def import_course_from_file(filename, user_id):
     except ArchiveException:
         remove(filename)
         raise ValueError("Invalid OLX archive, unable to extract.")
-    dirs = glob(join(tempdir, "*"))
-    if len(dirs) != 1:
-        rmtree(tempdir)
-        remove(filename)
-        raise ValueError("Invalid OLX archive, bad directory structure.")
-    import_course_from_path(dirs[0], user_id)
+    course_imported = False
```

Comment:
Are we only interested in course.xml in the root directory or immediate subdirectory? If recursive we should use os.walk. If there's only one place course.xml should be we should only check there, or add a comment if the intent is to be flexible with input (I don't know the xbundle format so I'm not sure about my reasoning here)